security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java

```diff
@@ -0,0 +1,348 @@
+		String wsOIGPassword = prop.getProperty("wsOIGPassword");
+
+		Hashtable<String, String> env = new Hashtable<>();
+		env.put(OIMClient.JAVA_NAMING_FACTORY_INITIAL, "weblogic.jndi.WLInitialContextFactory");
+		env.put(OIMClient.JAVA_NAMING_PROVIDER_URL, wsOIGNodeURL);
+		oimClient = new OIMClient(env);
+		oimClient.login(wsOIGUser, wsOIGPassword.toCharArray());
+
+		logger.log(Level.FINE, "OIM Client initialization complete.");
+	}
+
+	/**
+	 * Initiate a DB connection. Note that the web service needs to be deployed on a
+	 * WebLogic node which also has access to the jdbc/operationsDB datasource, also
+	 * known as oimOperationsDB. This datasource is typically deployed only on the
+	 * OIM and SOA nodes of a WebLogic cluster.
+	 * 
+	 * @return
+	 * @throws IOException
+	 * @throws SQLException
+	 * @throws NamingException
+	 */
+	private Connection getDBConnection() throws IOException, SQLException, NamingException {
+		logger.log(Level.FINE, "Initializing DB Datasource connection...");
+
+		Properties prop = new Properties();
+		InputStream configurationsFileInput = null;
+
+		String wlDomainDirectory = System.getProperty("domain.home");
+		configurationsFileInput = new FileInputStream(
+				java.nio.file.Paths.get(wlDomainDirectory, "config", "extendaccessws.properties").toString());
+		prop.load(configurationsFileInput);
+
+		try {
+			configurationsFileInput.close();
+		} catch (IOException e) {
+			logger.log(Level.WARNING, "Unable to close properties file.");
+		}
+
+		String wsExtendAccessNodeURL = prop.getProperty("wsExtendAccessNodeURL");
+		logger.log(Level.FINE, "wsExtendAccessNodeURL: {0}", wsExtendAccessNodeURL);
+
+		Hashtable<Object, String> env = new Hashtable<>();
+		env.put(OIMClient.JAVA_NAMING_FACTORY_INITIAL, "weblogic.jndi.WLInitialContextFactory");
+		env.put(OIMClient.JAVA_NAMING_PROVIDER_URL, wsExtendAccessNodeURL);
+		Context ctx = new InitialContext(env);
+		DataSource ds = (javax.sql.DataSource) ctx.lookup("jdbc/operationsDB");
+		Connection connection = ds.getConnection();
+
+		logger.log(Level.FINE, "DB Datasource connection initialization complete.");
+		return connection;
+	}
+
+	/**
+	 * Extend the access for accounts and entitlements based on previously saved DB
+	 * entries. The code will validate against a UUID token before access extension.
+	 * 
+	 * @return
+	 */
+	@GET
+	@Path("/extend/{userId}/{eaUUID}")
+	@Produces(MediaType.TEXT_PLAIN)
+	public String extendAccess(@PathParam("userId") String userId, @PathParam("eaUUID") String eaUUID) {
+		logger.log(Level.INFO, "Extending access for userId: {0}", userId);
+		logger.log(Level.FINE, "Extension token: {0}", eaUUID);
+
+		// default return string (in case of unexpected exceptions)
+		String restOutput = "Error: Access could not be extended.";
+
+		Connection dbConnection = null;
+		PreparedStatement preparedStatement = null;
+		ResultSet resultSet = null;
+
+		try {
+			// connect to OIM using the oimClient service
+			if (oimClient == null)
```

Comment:
Surround every block even single line blocks with curly braces. One famous Apple TLS bug was because of missing curly braces.

Reply:
Fixed.